logic/src/logic_wire.cpp

```diff
@@ -19,25 +19,29 @@
 
 // Definition of operator for WireInput (Allows it to be stored in maps).
 bool operator < (const WireInput &left, const WireInput &right) {
-	if( left.gateID < right.gateID ) {
+	if (left.gateID < right.gateID) {
 		return true;
-	} else if( left.gateID > right.gateID ) {
+	} else if (left.gateID > right.gateID) {
 		return false;
-	} else if( left.gateOutputID < right.gateOutputID ) {
+	} else if (left.gateOutputID < right.gateOutputID) {
 		return true;
-	} else return false;
+	}
+
+	return false;
 }
```

Comment:
This was just a formatting change, but this code should read:

```cpp
return left.gateID < right.gateID || (left.gateID == right.gateID && left.gateOutputID < right.gateOutputID)
```

That may not be the right logic, but at least this makes what is more clearly stated.